src/osd/PrimaryLogScrub.cc

```diff
@@ -546,8 +546,7 @@ void PrimaryLogScrub::scrub_snapshot_metadata(ScrubMap& scrubmap,
 
     ++num_digest_updates_pending;
     ctx->register_on_success([this]() {
-      dout(20) << "updating scrub digest " << num_digest_updates_pending << dendl;
-      if (--num_digest_updates_pending <= 0) {
+      if ((num_digest_updates_pending >= 1) && (--num_digest_updates_pending <= 0)) {
```

Comment:
should we need make the `num_digest_updates_pending` check same as what we have in quincy/master?

Reply:
@neha-ojha : do you mean '== 0' instead of the '<= 0'?
It's logically the same (because of the 1'st test).
I can change it to be text-compatible.